NiL.JS/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿using NiL.JS.Backward;
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace NiL.JS.Extensions
+{
+    public static class TypeExtensions
+    {
+        /// <summary>
+        /// Determines if the given type matches Task<>
+        /// </summary>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        public static bool IsTaskOf(this Type type)
```

Comment:
I suppose it should not be a global extension. Class `Type` is not related to public engine api. I think, type `Tools` is more suitable place for this function. And not as extensions, but as simple function.

Reply:
Moved this method to Tools